net/Kconfig

```diff
@@ -434,6 +434,31 @@ config NET_STARHUB
 endchoice # Node role
 endmenu # Network Topologies
 
+menu "Net Offload"
+
+config NET_OFFLOAD
```

Comment:
I don't like this name, it's similar to Zephyr ""Offload IP stack" , but the functionality is different

https://github.com/zephyrproject-rtos/zephyr/blob/main/subsys/net/ip/Kconfig#L61-L80

Reply:
In Nuttx, net offload just means that things originally processed in the Nuttx network stack can be offloaded to the lower layers, such as drivers or even hardware. Related functions such as GSO, LRO/GRO, checksum.

Reply:
But I don’t see all protocols offloaded to the lower layer/hardware. The current implementation seems that NET_OFFLOAD only abstracts the Scatter/Gatter for some protocols.

Reply:
Yes, the checksum offload has not been submitted yet

Reply:
what I mean, NET_OFFLOAD should bypass nuttx stack and offload all L3 protocols (TCP/UDP/ICMP/IGMP/ARP) to lower layer, but NIC technologies like GSO/GRO/LRO/TSO means segmentation Offloading.

Reply:
NET_SEGMENT_OFFLOAD is OK?

Reply:
NET_SEG_OFFLOAD ?

Reply:
OK

Reply:
Done